prov/sockets/src/sock_progress.c

```diff
@@ -220,6 +230,8 @@ static struct sock_pe_entry *sock_pe_acquire_entry(struct sock_pe *pe)
 		pe->num_free_entries--;
 		entry = pe->free_list.next;
 		pe_entry = container_of(entry, struct sock_pe_entry, entry);
+		/* Ring buffer must be empty */
+		assert(ofi_rbavail(&pe_entry->comm_buf) == pe_entry->comm_buf.size);
```

Comment:
I ended up pulling in this patch manually, and made the above mentioned changes.